### PR TITLE
xds: accept all forms of gkeClusterUrl and fix the 'x-goog-request-params' value

### PR DESCRIPTION
**Pull request by sanjaypujare**
As described here https://cloud.google.com/kubernetes-engine/docs/reference/rest we should support both `/v1/` and `/v1beta1/` and `/locations/` and `/zones/` in the GKE Cluster URL we read from the bootstrap file.

Also the location metadata header value should be `location=locations/$ZONE`

### CODE REVIEW

File: xds/src/main/java/io/grpc/xds/internal/certprovider/MeshCaCertificateProviderProvider.java [hunk 1]
```diff
@@ -69,7 +69,7 @@
   static final long RPC_TIMEOUT_SECONDS = 10L;
 
   private static final Pattern CLUSTER_URL_PATTERN = Pattern
-      .compile(".*/projects/(.*)/locations/(.*)/clusters/.*");
+      .compile(".*/projects/(.*)/(?:locations|zones)/(.*)/clusters/.*");
```
Comment (dapengzhang0) [inline xds/src/main/java/io/grpc/xds/internal/certprovider/MeshCaCertificateProviderProvider.java:72]: Is `(.*)` the same as `.*`? Seems you are not capturing groups.

Comment (sanjaypujare) [reply xds/src/main/java/io/grpc/xds/internal/certprovider/MeshCaCertificateProviderProvider.java:72]: `(.*)` is a capturing group and I am capturing only 2 groups (to get the project name and the cluster name). The only change here is from:

`locations`

to

`(?:locations|zones)`

And the initial `?:` makes it a [non-capturing group](https://stackoverflow.com/questions/3512471/what-is-a-non-capturing-group-in-regular-expressions).

Comment (dapengzhang0) [reply xds/src/main/java/io/grpc/xds/internal/certprovider/MeshCaCertificateProviderProvider.java:72]: > to get the project name and the cluster name

To get the cluster name, why not `/clusters/(.*)`? I might misunderstand the URI.

Comment (dapengzhang0) [reply xds/src/main/java/io/grpc/xds/internal/certprovider/MeshCaCertificateProviderProvider.java:72]: Oh, I see you are actually getting project name and zone name 
https://github.com/grpc/grpc-java/blob/master/xds/src/main/java/io/grpc/xds/internal/certprovider/MeshCaCertificateProviderProvider.java#L245-L246

Comment (sanjaypujare) [reply xds/src/main/java/io/grpc/xds/internal/certprovider/MeshCaCertificateProviderProvider.java:72]: Oops, sorry I made a mistake in my previous comment. I don't need the cluster name but just the project name and the location (or zone) name. And that's why use the first capturing group for the part following `/projects/` and the second one for the part following either `/locations/` or `/zones/` which is why I have the `(?:locations|zones)` expression.

The GkeClusterUrl is of the form: `https://container.googleapis.com/$VERSION/projects/$PROJECT/$LOCATIONS/$ZONE/clusters/$CLUSTER`

where $LOCATIONS is either `locations` or `zones`

**Review by dapengzhang0 — APPROVED** [review]

**Pull request merged**